code/game/objects/machinery/computer/nuke_disk_generator.dm

```diff
@@ -187,6 +195,26 @@
 	SSminimaps.remove_marker(src)
 	SSminimaps.add_marker(src, MINIMAP_FLAG_ALL, image('icons/UI_icons/map_blips_large.dmi', null, "[disk_color]_disk[current_timer ? "_on" : "_off"]", VERY_HIGH_FLOAT_LAYER))
 
+///Applies the overclock boost when colony power is active
+/obj/machinery/computer/nuke_disk_generator/proc/start_power_overclocking()
+	if(current_overclock_multiplier != 1) //Already activated!
+		return
+	current_overclock_multiplier = overclock_multiplier
+	if(current_timer)
+		deltimer(current_timer)
+		var/seconds_remaining = (seconds_elapsed SECONDS) - (segment_time*completed_segments)
+		current_timer = addtimer(CALLBACK(src, PROC_REF(complete_segment)), (segment_time-seconds_remaining)/overclock_multiplier, TIMER_STOPPABLE)
```

Comment:
Couldnt you also use timeleft instead of this math? Any particular reason you did it this way?